src/transactionsFactories/smartContractTransactionsFactory.spec.ts

```diff
@@ -107,6 +109,202 @@ describe("test smart contract transactions factory", function () {
         assert.deepEqual(executeDraft, abiExecuteDraft);
     });
 
+    it("should create draft transaction for execute and transfer native token", async function () {
+        const sender = Address.fromBech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th");
+        const contract = Address.fromBech32("erd1qqqqqqqqqqqqqpgqhy6nl6zq07rnzry8uyh6rtyq0uzgtk3e69fqgtz9l4");
+        const func = "add";
+        const gasLimit = 6000000;
+        const args = [new U32Value(7)];
+        const egldAmount = new BigNumber("1000000000000000000");
+
+        const executeDraft = factory.createTransactionForExecute({
+            sender: sender,
+            contract: contract,
+            functionName: func,
+            gasLimit: gasLimit,
+            args: args,
+            nativeTransferAmount: egldAmount
+        });
+        const abiExecuteDraft = abiAwareFactory.createTransactionForExecute({
+            sender: sender,
+            contract: contract,
+            functionName: func,
+            gasLimit: gasLimit,
+            args: args,
```

Comment:
For the sake of the test (abi-aware), you can have:

```
args: [7]
```

And then, for the one above, `args: [new U32Value(7)]` (symmetry).

Reply:
Done!